### DISCUSSION TITLE
Add translations to EDI\Analyser

### DISCUSSION THREAD

**Discussion by mbrans-plieger** [php-edifact/edifact#128] [Ideas]
Working with users who don't all have a perfect knowledge of the English language, it would be great if the (very detailed) response of the EDI\Analyser could have an additional parameter to output the details in another language. At the moment I would really need German but Dutch, Spanish and French would also be great.
For sure I'm willing to participate in setting up the translation tables (FE via CrowdIn).

**Comment by sabas** [comment]
@mbrans-plieger hi, do you mean you'd wish to translate the UN/EDIFACT directories?
https://github.com/php-edifact/edifact-mapping
https://unece.org/trade/uncefact/unedifact/download
I wonder how it would be feasible since there are a lot of directories...

Add: technically loading a different directory is already possible ;)

**Comment by mbrans-plieger** [comment]
Hi Stefana,

I meant the EDI\Analyser.
Currently the output is like this:

UNA:+.? \'
UNB - InterchangeHeader
  (To start, identify and specify an interchange)
  [0] UNOC,3
      unb1 - syntaxIdentifier
      Syntax identifier
    [0] UNOC
        id: unb11 - syntaxIdentifier
        Syntax identifier
        type: a
        required: true
        length: 4
    [1] 3
        id: unb12 - syntaxVersionNumber
        Syntax version number
        type: n
        required: true
        length: 1
  [1] 8711238000015,14
      unb2 - interchangeSender
      Interchange sender
    [0] 8711238000015
        id: unb21 - senderIdentification
        Sender identification
        type: an
        maxlen: 35
        required: true
    [1] 14
        id: unb22 - identificationCodeQualifier
        Identification code qualifier
        type: an
        maxlen: 4
  [2] 4063718000003,14
      unb3 - interchangeRecipient
      Interchange recipient
    [0] 4063718000003
        id: unb31 - recipientIdentification

        type: an
        maxlen: 35
        required: true
    [1] 14
        id: unb32 - identificationCodeQualifier
        Identification code qualifier
        type: an
        maxlen: 4
  [3] 240222,1454
      unb4 - dateTimePreparation
      Date Time of preparation
    [0] 240222
        id: unb41 - date

        type: n
        required: true
        length: 6
    [1] 1454
        id: unb42 - time

        type: n
        required: true
        length: 4
  [4] 102
      unb5 - interchangeControlReference


I use it in this way:
            $filePath = $ediMessage['file_path'];
            $reader = new Reader($filePath);
            $parser = new Parser($filePath);
            $segments = $parser->getRawSegments();
            $parsed = $parser->get();
            $analyser = new Analyser();
            $mapping = new MappingProvider('D95B');
            $analyser->loadSegmentsXml($mapping->getSegments());
            $analyser->loadMessageXml($mapping->getMessage('coparn'));
            $analyser->loadCodesXml($mapping->getCodes());
            $analyser->directory = 'D95B';
            $analyserResult = $analyser->process($parsed, $segments);

Would be great if I can add an optional parameter somewhere to set an output language so all the detailed explanation would be in for example German.

Hope this clarifies the intention.

Thanks,

Micha





Werktijden/office hours: 7:00 - 15:30


Van: Stefano ***@***.***>
Verzonden: donderdag 29 februari 2024 08:45
Aan: php-edifact/edifact ***@***.***>
CC: Micha Brans ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [php-edifact/edifact] Add translations to EDI\Analyser (Discussion #128)

U ontvangt niet vaak e-mail van ***@***.******@***.***>. Meer informatie over waarom dit belangrijk is<https://aka.ms/LearnAboutSenderIdentification>

@mbrans-plieger<https://github.com/mbrans-plieger> hi, do you mean you'd wish to translate the UN/EDIFACT directories?
https://github.com/php-edifact/edifact-mapping
https://unece.org/trade/uncefact/unedifact/download
I wonder how it would be feasible since there are a lot of directories...

-
Reply to this email directly, view it on GitHub<https://github.com/php-edifact/edifact/discussions/128#discussioncomment-8627133>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ASJMKFJZEXRWIY7YDVT2HZDYV3OBJAVCNFSM6AAAAABD7MXYDSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DMMRXGEZTG>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>
Micha Brans
Senior DevOps Engineer
+31 418 682693 	 +31 6 28878578

Koxkampseweg 6, 5301 KK Zaltbommel | +31 418 682682

**Reply by sabas** [reply]
Hello Micha,
to clarify, apart from id / syntax / type / length which come from the code, the content like "identificationCodeQualifier Identification code qualifier" comes from this xml
https://github.com/php-edifact/edifact-mapping/blob/master/src/Mapping/D95B/segments.xml#L1804
